frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,156 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
-    genMenuItem(_("Leave screen as-is"), "screensaver_type", "disable", nil, true),
-    -- separator
-    {
-        text = _("Add message to screensaver"),
-        checked_func = function()
-            return G_reader_settings:isTrue("screensaver_show_message")
-        end,
-        callback = function()
-            G_reader_settings:toggle("screensaver_show_message")
-        end,
-        separator = true,
-    },
-    -- separator
     {
-        text = _("Settings"),
+        text = _("Wallpaper"),
         sub_item_table = {
+            genMenuItem(_("Show book cover on sleep screen"), "screensaver_type", "cover", hasLastFile),
+            genMenuItem(_("Show custom image on sleep screen"), "screensaver_type", "image_file"),
+            genMenuItem(_("Use random image from folder on sleep screen"), "screensaver_type", "random_image"),
+            genMenuItem(_("Show document cover on sleep screen"), "screensaver_type", "document_cover"),
+            genMenuItem(_("Show reading progress on sleep screen"), "screensaver_type", "readingprogress", isReaderProgressEnabled),
+            genMenuItem(_("Show book status on sleep screen"), "screensaver_type", "bookstatus", hasLastFile),
+            genMenuItem(_("Leave screen as-is"), "screensaver_type", "disable", nil, true),
+            separator = true,
             {
-                text = _("Screensaver folder"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFolder()
+                text = _("Border fill"),
+                enabled_func = function()
+                    return G_reader_settings:readSetting("screensaver_type") == "cover"
+                            or G_reader_settings:readSetting("screensaver_type") == "random_image"
+                            or G_reader_settings:readSetting("screensaver_type") == "image_file"
+                            or G_reader_settings:readSetting("screensaver_type") == "document_cover"
                 end,
+                sub_item_table = {
+                    genMenuItem(_("Black"), "screensaver_img_background", "black"),
+                    genMenuItem(_("White"), "screensaver_img_background", "white"),
+                    genMenuItem(_("Current state"), "screensaver_img_background", "none", nil, true),
+                    -- separator
+                        {
+                            text_func = function()
+                                local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
+                                if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
+                                    return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
+                                end
+                                return _("Stretch cover to fit screen")
```

Comment:
(When playing, I was expecting that enabling this one, the 3 others (Black/White/Current) would be disabled. But no, they can still be enabled, and may have some effect because of the stretching % limit.)